torch/testing/_internal/common_device_type.py

```diff
@@ -1337,7 +1337,7 @@ def wrap_fn(self, *args, **kwargs):
 
 # Skips a test on CUDA if cuSparse generic API is not available
 def skipCUDAIfNoCusparseGeneric(fn):
-    return skipCUDAIf(not TEST_CUSPARSE_GENERIC, "cuSparse Generic API not available")(fn)
+    return skipCUDAIf(not TEST_WITH_ROCM and not TEST_CUSPARSE_GENERIC, "cuSparse Generic API not available")(fn)
```

Comment:
Let's please not mix this in, because we need to be able to keep these separate.